### ISSUE TITLE
system_groups and root_account modify the same thing

### ISSUE THREAD

**Issue by gardouille** [debops/debops#344]
I guess i have some kind of loop.

[Enforce root account task](https://github.com/debops/debops/blob/master/ansible/roles/debops.root_account/tasks/main.yml#L18) (from debops.root_account) and [Add specified UNIX accounts… task](https://github.com/debops/debops/blob/master/ansible/roles/debops.system_groups/tasks/main.yml#L18) (from debops.system_groups) both modify the same thing on my configuration.

### First run
``` yaml
TASK [debops.system_groups : Add specified UNIX accounts to system groups if present] **************************************************
changed: [mynode.domain.org] => (item={'groups': u'admins,adm,staff', 'append': True, 'name': u'root'})
```
  * And on mynode.domain.org, root's groups are :
``` sh
id root
uid=0(root) gid=0(root) groups=0(root),4(adm),50(staff),998(admins)
```
  * `/etc/ansible/facts.d/core.fact` output :
``` yaml
{
  "admin_groups": [
    "admins"
  ],
  "admin_private_email": [
    "root"
  ],
  "admin_public_email": [
    "root@domain.org"
  ],
  "admin_users": [
    "root"
  ],
  "ansible_controllers": [
    "129.20.79.142"
  ],
  "cache_valid_time": "86400",
  "distribution": "Debian",
  "distribution_release": "stretch",
  "domain": "domain.org",
  "fqdn": "mynode.domain.org",
  "homedir_umask": "0027",
  "keyserver": "hkp://pool.sks-keyservers.net",
  "unsafe_writes": false
}
```

### Second run
``` yaml
TASK [debops.root_account : Enforce root system account] *******************************************************************************
changed: [mynode.domain.org]
```
  * Then, root's groups :
``` sh
id root
uid=0(root) gid=0(root) groups=0(root)
```
  * `/etc/ansible/facts.d/core.fact` output :
``` yaml
{
  "admin_groups": [
    "admins"
  ],
  "admin_private_email": [
    "root"
  ],
  "admin_public_email": [
    "root@domain.org"
  ],
  "admin_users": [
    "root"
  ],
  "ansible_controllers": [
    "129.20.79.142"
  ],
  "cache_valid_time": "86400",
  "distribution": "Debian",
  "distribution_release": "stretch",
  "domain": "domain.org",
  "fqdn": "mynode.domain.org",
  "homedir_umask": "0027",
  "keyserver": "hkp://pool.sks-keyservers.net",
  "unsafe_writes": false
}
```

### Configuration
* The vars i override for debops.root_account :
``` yaml
root_account__password_update: False
root_account__generate_ssh_key: False
```
* And for debops.system_groups :
``` yaml
system_groups__admins_sudo_nopasswd: False
system_groups__list:
  - name: 'admins'
    sudoers: |
      USER1 ALL=(ALL:ALL)  ALL
      USER2 ALL=(ALL:ALL)  ALL
      USER3 ALL=(ALL:ALL)  ALL
```

Did i miss some value or anything else ?

**Comment by drybjed** [comment]
I suspect that somewhere in inventory you have `system_groups__*` variable that references the `root` account in a wrong way. Review the inventory, for example with `grep` to find all `system_groups__` variables.

**Comment by drybjed** [comment]
It might also be due to `root` being in `ansible_local.core.admin_users` fact variable. Check the output of `/etc/ansible/facts.d/core.fact` to see if it returns `root` as one of the admin accounts.

**Comment by gardouille** [comment]
@drybjed I just update the first comment, it seems root is an admin_user but not in admin_group.
The two `system_groups__*` vars from the first comment are the only ones this node is able to see (from inventory/group_vars/all/debops.yml).

Also, i don't have any var for `core__*`. And i check all **root** occurences in inventory files and it's only set to :
* Give sudoers permissions
* As option for logrotate (eg. create 0664 root root)
* `root_account__*` vars defined in the first comment
* `unattended_upgrades__mail_to: [ 'root@localhost' ]`
* `fail2ban_destemail: 'root@localhost'`

**Comment by gardouille** [comment]
I also have `ansible_user=root` (3 admins authenticated by LDAP). Everything is fixed with #345 :)

In a long term, kerberos, `ansible_become_pass`,… should be a better solution.

Thanks ;)
